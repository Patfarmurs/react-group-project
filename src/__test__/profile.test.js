import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';

import Profile from '../pages/profile';

const mockStore = configureStore([]);

describe('Profile component', () => {
  test('renders correctly', () => {
    const store = mockStore({
      rocket: { rockets: [{ id: 1, reserved: true }] },
      mission: { missions: [{ id: 1, joined: true }] },
    });

    const { container } = render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
