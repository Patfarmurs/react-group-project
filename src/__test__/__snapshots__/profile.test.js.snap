// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component renders correctly 1`] = `
<main
  class="mainProfile"
>
  <section
    class="reservedRocket"
  >
    <h2>
      My Rockets
    </h2>
    <ul>
      <li
        class="resRocket"
      >
        <h2 />
        <p />
      </li>
    </ul>
  </section>
  <section
    class="missionJoined"
  >
    <h2>
      My Missions
    </h2>
    <ul>
      <li
        class="misJoined"
      >
        <h2 />
        <p />
      </li>
    </ul>
  </section>
</main>
`;
